docs/source/posts/2024/2024-07-01-week-4-robin.rst

```diff
@@ -0,0 +1,74 @@
+
+Things I did in Week 4
+----------------------
+
+1) **Chunking tutorials and documentation**
+
+Earlier, only files fitting the context window of the embedding model were upserted. This was because otherwise, we'd have to split the file in half and lose the overall context. This will lead to information loss and retrieval will be messy. Now, I decided I'd upsert everything by splitting information properly. By "properly", what I mean is it won't be a random split, and there'll be logical reasoning behind every chunk.
+
+This area is still actively studied, and the whole concept is to find ideal chunks which are self-sufficient and contain the most information. This `notebook <https://github.com/FullStackRetrieval-com/RetrievalTutorials/blob/main/tutorials/LevelsOfTextSplitting/5_Levels_Of_Text_Splitting.ipynb>`_ details 6 different approaches, I read through them and some of their associated literature and decided we'll use `Recursive Character Text Splitting` and `Document Specific Splitting` for now. There is no major reason for this, I just felt it'll work well for now (a reasoning-backed approach will come in a few weeks). There is a lot of experimentation we could do here, a better chunking will result in better ``references`` generation and so on.
+
+So this is our current process
+    - if normal function/class definition: no splitting, chunk as it is.
+    - if rst files, use the ``rst parser`` and split them with a chunk size of ~8000 tokens (max llama could take)
+    - if tutorial, try chunking as it is, if not possible split at 8000 tokens.
+
+Function/class definitions are generally under 8000 so I've not done explicit checks for now, the model will trim the remaining if longer (I found some long classes later).
+
+2) **Move colab files to a proper Python script**
+
+I did all the upsertion and experiments on colab. It is messy and can't be used in production. We need a one-click approach to upsertion. Something like point to `fury` directory and it should do everything. So I took everything and made a script.
+
+One of my key goals is to separate core application logic from LLMs/Database providers. We should be able to swap them as needed without much fuss. I'll talk more about this in week 5.
+
+3) **Taking the bot public!**
+
+The whole point of making the bot is to help improve the productivity of FURY developers. So I decided to take it public on `this discord server <https://discord.gg/NVkE6Qd2bZ>`_. You could use it today! (actually, you could've used it from the 20th of last month, this blog got delayed😢)
+
+I'll observe what people are asking and then iterate towards making the bot better in that area. I think it'll be better than making the bot good on what I believe is the best.
+
+4) **Minor bugfixes and stuff**
```

Comment:
Describe the specific bugs you encountered and how you fixed them.

Reply:
Hi Wachiou, thank you for the review. I've made the changes :+1: